packages/accounts-2fa/2fa-server.js

```diff
@@ -96,37 +96,44 @@ Meteor.methods({
     }
 
     Meteor.users.update(
-      { username },
+      { _id: user._id },
       {
         $set: {
           'services.twoFactorAuthentication': {
             ...twoFactorAuthentication,
-            type: 'otp',
-          },
-        },
+            type: 'otp'
+          }
+        }
       }
     );
   },
   disableUser2fa() {
-    const user = Meteor.user();
+    const userId = Meteor.userId();
 
-    if (!user) {
+    if (!userId) {
       throw new Meteor.Error(400, 'No user logged in.');
     }
 
     Meteor.users.update(
-      { username: user.username },
+      { _id: userId },
       {
         $unset: {
-          'services.twoFactorAuthentication': 1,
-        },
+          'services.twoFactorAuthentication': 1
+        }
       }
     );
   },
   has2faEnabled(selector) {
-    if (!Meteor.user()) {
+    check(selector, Match.Maybe(Match.OneOf(String, Object)));
+    const userId = Meteor.userId();
+    if (!userId) {
       throw new Meteor.Error(400, 'No user logged in.');
     }
+
+    if (!selector) {
+      selector = { _id: userId };
```

Comment:
Made selector optional. If none is passed, then it will look for the current user.